text/0000-project-descriptor.md

```diff
@@ -0,0 +1,324 @@
+# Meta
+[meta]: #meta
+- Name: Project Descriptor
+- Start Date: 2019-06-11
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: https://github.com/buildpack/rfcs/pull/16
+
+# Summary
+[summary]: #summary
+
+This is a proposal for a new file, `project.toml`, containing configuration for apps, services, functions, and buildpacks.
+
+# Motivation
+[motivation]: #motivation
+
+This proposal is meant to address all of the following issues:
+
+* Build metadata (i.e. information about the code being built)
+* [Monorepos](https://github.com/buildpack/spec/issues/39)
+* [Application descriptor](https://github.com/buildpack/spec/issues/44)
+* [Ignoring files](https://github.com/buildpack/pack/issues/210)
+* Multiple images (i.e. one build, multiple images as output)
+* Custom launch configuration
+* Codified buildpacks
+
+# What it is
+[what-it-is]: #what-it-is
+
+Terminology:
+
+* **project**: a repository containing source code for an app, service, function, buildpack or a monorepo containing any combination of those.
+* **image**: the output of a running a buildpack(s) against a project
+
+The target personas for this proposal is buildpack users who need to enrich or configure buildpack execution. The elements in  `project.toml` support many different use cases including three top level tables:
+
+- `[project]`: (optional) defines configuration for a project
+- `[[images]]`: (optional) defines configuration for an image
+- `[metadata]`: (optional) metadata about the repository
+- `[extensions]`: (optional) extensions to the spec
+
+Here is an overview of the complete schema:
+
+```toml
+[project]
+id = "<string>"
+name = "<string>"
+version = "<string>"
+authors = ["<string>"]
+documentation = "<url>"
+license = "<string>"
+source = "<url>"
+include = ["<string>"]
+exclude = ["<string>"]
+
+[[images]]
+ref = "<string>"
```

Comment:
Small issue with `ref` vs `tag`: only tags are valid in this field, and other kinds of ref are not valid.
E.g., `gcr.io/example/myapp@sha256:2342ab...` is not valid, because it's not possible to know the digest ahead of time.

Reply:
i'm fine with `tag` then